extensions/mssql/src/objectManagement/ui/databaseDialog.ts

```diff
@@ -396,6 +419,208 @@ export class DatabaseDialog extends ObjectManagementDialogBase<Database, Databas
+			file.fileNameWithExtension
+		];
+	}
+
+	private async onAddDatabaseFilesButtonClicked(button: azdata.ButtonComponent): Promise<void> {
+		// Open file dialog to create file
+		const result = await this.openDatabaseFileDialog(button);
+		if (!isUndefinedOrNull(result)) {
+			this.objectInfo.files?.push(result);
+			var newData = this.objectInfo.files?.map(file => {
+				return this.convertToDataView(file);
+			});
+			await this.setTableData(this.databaseFilesTable, newData, DefaultMaxTableRowCount)
+		}
+	}
+
+	private async onEditDatabaseFilesButtonClicked(button: azdata.ButtonComponent): Promise<void> {
+		if (this.databaseFilesTable.selectedRows.length === 1) {
+			const result = await this.openDatabaseFileDialog(button);
+			if (!isUndefinedOrNull(result)) {
+				this.objectInfo.files[this.databaseFilesTable.selectedRows[0]] = result;
+				var newData = this.objectInfo.files?.map(file => {
+					return this.convertToDataView(file);
+				});
+				await this.setTableData(this.databaseFilesTable, newData, DefaultMaxTableRowCount)
+			}
+		}
+	}
+
+	/**
+	 * Removes the selected database file from the table
+	 */
+	private async onRemoveDatabaseFilesButtonClicked(): Promise<void> {
+		if (this.databaseFilesTable.selectedRows.length === 1) {
+			this.objectInfo.files?.splice(this.databaseFilesTable.selectedRows[0], 1);
+			var newData = this.objectInfo.files?.map(file => {
+				return this.convertToDataView(file);
+			});
+			await this.setTableData(this.databaseFilesTable, newData, DefaultMaxTableRowCount)
+		}
+	}
+
+	/**
+	 * Validate the selected row to enable/disable the remove button
+	 * @returns true if the remove button should be enabled, false otherwise
+	 */
+	protected override removeButtonOnRowSelected(): boolean {
+		let isEnabled = true;
+		const selectedRowId = this.objectInfo.files[this.databaseFilesTable.selectedRows[0]].id;
+		// Cannot delete a row file if the Id is 1.
+		if (this.databaseFilesTable.selectedRows.length === 1 && selectedRowId === 1) {
+			isEnabled = false;
+		}
+		// Cannot remove a log file if there are no other log files
+		else if (this.objectInfo.files[this.databaseFilesTable.selectedRows[0]].type === this.viewInfo.fileTypesOptions[1]) {
+			isEnabled = false;
+			this.objectInfo.files.forEach(file => {
+				if (file.id !== selectedRowId && file.type === this.viewInfo.fileTypesOptions[1]) {
+					isEnabled = true;
+				}
+			});
+		}
+		return isEnabled;
+	}
+
+	private async openDatabaseFileDialog(button: azdata.ButtonComponent): Promise<DatabaseFile> {
+		const defaultFileSizeInMb: number = 8
+		const defaultFileGrowthInMb: number = 64
+		const defaultFileGrowthInPercent: number = 10;
+		const defaultMaxFileSizeLimitedToInMb: number = 100;
+		const selectedFile = this.databaseFilesTable.selectedRows !== undefined ? this.objectInfo.files[this.databaseFilesTable?.selectedRows[0]] : undefined;
+		if (!isUndefinedOrNull(selectedFile) && selectedFile.type === this.viewInfo.fileTypesOptions[2]) {
```

Comment:
What is this check doing? Should you really be using an indexed value from fileTypesOptions for comparison instead of checking against a specific constant? Relying on this array to stay in exactly the same order seems risky.